src/lib/util/DynamicSizeWrapper/DynamicSizeWrapper.tsx

```diff
@@ -0,0 +1,49 @@
+import { ReactNode, useEffect, useRef, useState } from 'react';
+
+//////////////////////////
+// Dynamic Size Wrapper //
+//////////////////////////
+
+export interface DynamicSizeWrapperProps {
+	className?: string;
+	children: ReactNode;
+}
+
+export const DynamicSizeWrapper = ({
+	className,
+	children,
+}: DynamicSizeWrapperProps) => {
+	const ref = useRef<HTMLDivElement>();
+	const [breakpoints, setBreakpoints] = useState<string | undefined>();
+
+	useEffect(() => {
+		if (!ref) {
+			return;
+		}
+
+		const resizeObserver = new ResizeObserver((event) => {
+			const width = event[0].contentBoxSize[0].inlineSize;
+			const breakpointWidths = Array.from(
+				{ length: Math.ceil(width / 50) },
+				(_, i) => i * 50,
+			);
+			setBreakpoints(breakpointWidths.join('-'));
```

Comment:
This feels unnecessarily granular to me. How many different sizes does the app need to account for? Most really only have a few. Something like, small/medium/large?

Could we just limit to 3 sizes and rather than use a data-attribute just put a class on the main container? `container--small`

Reply:
The problem I generally have with `small`, `medium`, `large`, is that it's rare for elements to naturally fit into `small`, `medium`, and `large` buckets. 

For example, you might have a row of `Card` elements which look good up until `500px`, at which point you'd want to switch them to a column layout. If you have `small = 300`, `medium = 700`, your only choice is to switch to a column layout at `700px`, which means there's a `200px` gap where the `Card` elements will look unnaturally stretched out when they could have instead stayed arranged as a row. The bigger the gaps between your breakpoint definitions, the more you're going to run into cases where the layout has no choice but to adapt earlier than it needs to. You can add more values, like `mobile-s`, `mobile-m`, etc. but I think at that point you may as well just use increments of `50` to reduce complexity.

tl;dr: imo you get a much nicer and consistent UI if you dial in breakpoints to the width that elements actually break at, rather than rounding breakpoints into buckets, but could just be my pedantic design-brain.

Reply:
Well, that depends on the system the design team is using. Do you have a design right now that you're matching to that requires you to split on numerous values that aren't nicely predefined? It's not a huge deal but it definitely seems overly complex for any design I've ever had to implement.